server/src/test/java/org/elasticsearch/action/ingest/SimulatePipelineRequestParsingTests.java

```diff
@@ -211,6 +211,38 @@ public void testParseWithProvidedPipeline() throws Exception {
         assertThat(actualRequest.pipeline().getProcessors().size(), equalTo(numProcessors));
     }
 
+    public void testParseWithNestedProcessors() throws Exception {
+        Map<String, Object> requestContent = new HashMap<>();
+        Map<String, Object> pipelineConfig = new HashMap<>();
+        List<Map<String, Object>> processors = new ArrayList<>();
+        Map<String, Object> processorConfig = new HashMap<>();
+        List<Map<String, Object>> docs = new ArrayList<>();
+        requestContent.put(Fields.DOCS, docs);
+        String fieldName = randomAlphaOfLengthBetween(1, 10);
+        String fieldValue = randomAlphaOfLengthBetween(1, 10);
+        Map<String, Object> doc = new HashMap<>();
+        doc.put(Fields.SOURCE, Collections.singletonMap(fieldName, fieldValue));
```

Comment:
Prefer `Map.of(...)` rather than `Collections.singletonMap(...)` where possible --  there's this line and another  a few lines below.